### PR TITLE
feat(json_fmt): helm changes to enable json support

### PR DESCRIPTION
**Pull request by abhilashshetty04**
base.loging.format= json|pretty|compact . 

ansi colour can be disabled also. Its enabled by default.

### CODE REVIEW

File: chart/values.yaml [hunk 1]
```diff
@@ -51,6 +51,9 @@ base:
   default_req_timeout: 5s
   # -- Cache timeout for core agent & diskpool deployment
   cache_poll_period: 30s
+  logFormat:
+    jsonEnabled: false
+    ansiColorsEnabled: true
```
Comment (tiagolobocastro) [inline chart/values.yaml]: how about an enum?
And maybe combine with other log stuff such as silence level

```suggestion
  logging:
    format: pretty|compact|json
    color: true
    silenceLevel:
```

Comment (abhilashshetty04) [reply chart/values.yaml]: @tiagolobocastro , Initially thought about it. However, we use `FmtStyle::Compact ` only once while starting a csi node cmd line (freeze/unfreeze) service. Everywhere else its `FmtStyle::Pretty` by default. 

With my changes we keep the same behaviour. If someone wants to have json stdout they enable it or else its whatever style each service defaults to now.

Comment (tiagolobocastro) [reply chart/values.yaml]: Yes but if we want to change it then it becomes a ambiguous:
jsonEnabled: true
compactEnabled: true

File: chart/values.yaml [hunk 2]
```diff
@@ -51,6 +51,11 @@ base:
   default_req_timeout: 5s
   # -- Cache timeout for core agent & diskpool deployment
   cache_poll_period: 30s
+  logging:
+    # -- Valid values for format are pretty, json and compact
+    format: pretty
+    color: true
+    silenceLevel:
```
Comment (Abhinandan-Purkait) [inline chart/values.yaml:67]: Where are we setting the silenceLevel? Did not see that on the template.

**Review by Abhinandan-Purkait — APPROVED** [review]

File: chart/values.yaml (hunk 2 above)
Comment (abhilashshetty04) [reply chart/values.yaml:67]: Not sure, Installation works post removing silenceLevel also. In develop,  `-  logSilenceLevel: `, Doesnt have any further key/val also. Seems like an incomplete config.

@niladrih , @tiagolobocastro any ideas?

Comment (tiagolobocastro) [reply chart/values.yaml:67]: It's empty by default, meaning it's using container defaults. @abhilashshetty04 you need to modify the uses of the silenceLevel on every yaml, example: https://github.com/openebs/mayastor-extensions/blob/81ede2bbdbce5fd8ad64ad762390347529883733/chart/templates/mayastor/agents/core/agent-core-deployment.yaml#L67

Comment (abhilashshetty04) [reply chart/values.yaml:67]: Thanks :0

File: chart/values.yaml [hunk 3]
```diff
@@ -323,7 +328,7 @@ csi:
       keep_alive_tmo: ""
     # -- The kubeletDir directory for the csi-node plugin
     kubeletDir: /var/lib/kubelet
-    pluginMounthPath: /csi
+    pluginMountPath: /csi
```
Comment (tiagolobocastro) [inline chart/values.yaml]: Please move this typo change to separate commit.
Also @niladrih we might need to plugin upgrade code to support this?

Comment (tiagolobocastro) [reply chart/values.yaml]: I'd suggest to remove it from the PR and add both typo fix and plugin code in another PR

Comment (abhilashshetty04) [reply chart/values.yaml]: Sure @tiagolobocastro , But I thought we dont need code change for this. Its just changes in values.yaml and its uses.

Comment (tiagolobocastro) [reply chart/values.yaml]: Yes we don't need this change, let's remove it.

File: chart/templates/mayastor/agents/core/agent-core-deployment.yaml [hunk 4]
```diff
@@ -59,14 +59,16 @@ spec:
             - "--events-url=nats://{{ .Release.Name }}-nats:4222"{{ end }}
             {{- if not .Values.agents.ha.enabled }}
             - "--disable-ha"{{ end }}
+            - "--fmt-style={{ include "logFormat" . }}"
```
Comment (tiagolobocastro) [inline chart/templates/mayastor/agents/core/agent-core-deployment.yaml:62]: you seem to have undone the macro calling?

Comment (abhilashshetty04) [reply chart/templates/mayastor/agents/core/agent-core-deployment.yaml:62]: @tiagolobocastro , What do you mean?

Btw this is how it will be passed. 
- "--ansi-colors=true"
- "--fmt-style=pretty"

tpl fn will invoke error if .Values.base.logging.format is not json | pretty | compact

File: chart/values.yaml (hunk 3 above)
Comment (niladrih) [reply chart/values.yaml]: This will need an upgrade change, yes.

Comment (abhilashshetty04) [reply chart/values.yaml]: Reverted the changes

Comment (niladrih) [reply chart/values.yaml]: I've raised PR with the upgrade changes required for both (1) the logging values changes, (2) the plugin mount path typo change.
https://github.com/openebs/mayastor-extensions/pull/443.

File: chart/templates/mayastor/agents/core/agent-core-deployment.yaml (hunk 4 above)
Comment (tiagolobocastro) [reply chart/templates/mayastor/agents/core/agent-core-deployment.yaml:62]: [17:15] Tiago Castro

I thought I had in the PR previously passing both json and color options using the macro

[17:16] Tiago Castro

        args:

           - "--fmt-style={{ include "logFormat" . }}"

           - "--ansi-colors={{ .Values.base.logging.color }}"

versus

        args:

           {{ include "logOptions" . }}

**Review by tiagolobocastro — APPROVED** [review]

File: chart/templates/mayastor/agents/core/agent-core-deployment.yaml (hunk 4 above)
Comment (abhilashshetty04) [reply chart/templates/mayastor/agents/core/agent-core-deployment.yaml:62]: This was the latest commit with macro call. its onlt for validating and passing fmt-style. ansi-colors is passed as usual assignment

**Comment by abhilashshetty04** [comment]
bors merge

**Comment by abhilashshetty04** [comment]
bors try

**Comment by abhilashshetty04** [comment]
bors merge
